"""
2. Создайте собственный класс-исключение, обрабатывающий ситуацию деления на нуль.
Проверьте его работу на данных, вводимых пользователем. При вводе пользователем
нуля в качестве делителя программа должна корректно обработать эту ситуацию
и не завершиться с ошибкой.

"""

class OwnError(Exception):
    def __init__(self, txt):
        self.txt = txt

digit1 = int(input("Введите делимое: "))
digit2 = int(input("Введите делитель: "))

try:
    if digit2 == 0:
        raise OwnError("На ноль делить нельзя!")
except ValueError:
    print("Вы ввели не число")
except OwnError as err:
    print(err)
else:
    print(f"Все хорошо. Результат деления: {digit1/digit2}")